### PR TITLE
feat(components): add app header and layout

### PR DESCRIPTION
**Pull request by zouxuoz**

### CODE REVIEW

File: client/components/AppHeader/AppHeader.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  justify-content: center;
+}
+
+.app-header__content {
+  display: flex;
+  max-width: 1150px;
+  min-width: 960px;
+}
+
+.app-header__logo {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: Move it to `AppHeaderLogo.css` and import it in the `AppHeaderItem.js` component.

File: client/components/AppHeader/AppHeader.css [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  fill: var(--whiteColor);
+}
+
+.app-header__nav {
+  align-items: center;
+  display: flex;
+  height: 65px;
+}
+
+.app-header__link {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: Move it to `AppHeaderItem.css` and import it in the `AppHeaderItem.js` component.

File: client/components/AppHeader/AppHeader.css [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  text-decoration: none;
+  transition: color .2s ease;
+  white-space: nowrap;
+}
+
+.app-header__link:hover {
+  color: var(--whiteColor);
+}
+
+.app-header__item:not(:last-child) {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: Can we avoid `:not(:last-child)` and use the sibling selector:

```
.app-header__item {
  & + .app-header__item {
    margin-left: 30px;
  }
}

File: client/components/AppHeader/AppHeader.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, { Component, PropTypes } from 'react';
+import AppHeaderLogo from './AppHeaderLogo';
+import AppHeaderItem from './AppHeaderItem';
+import './AppHeader.css';
+
+class AppHeader extends Component {
+  render () {
+    return (
+      <header className="app-header">
+        <div className="app-header__content">
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.js]: Don't like the name `app-header__content`. May be `app-header__wrapper` or `app-header__container` (more like that variant)?

File: client/components/AppHeader/AppHeaderItem.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+class AppHeaderItem extends Component {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeaderItem.js]: Refactor it to a stateless functional component:

```
export default AppHeaderItem = () => (
);
```

File: client/styles/pallete.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+:root {
+  --whiteColor: #fff;
```
Comment (rosendi) [inline client/styles/pallete.css]: WTF? :-) We can use `white` color from the ancient times :-)

File: client/components/AppHeader/AppHeader.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+.app-header__logo {
+  align-items: center;
+  display: flex;
+  fill: color(var(--whiteColor) a(60%));
+  margin-right: 24px;
+}
+
+.app-header__logo:hover {
+  fill: var(--whiteColor);
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: See the last comment in `pallete.css` :-)

File: client/components/AppHeader/AppHeader.css [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+.app-header__content {
+  display: flex;
+  max-width: 1150px;
+  min-width: 960px;
+}
+
+.app-header__logo {
+  align-items: center;
+  display: flex;
+  fill: color(var(--whiteColor) a(60%));
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: See the last comment in `pallete.css` :-)

File: client/styles/pallete.css [hunk 9]
```diff
@@ -0,0 +1,4 @@
+:root {
```
Comment (rosendi) [inline client/styles/pallete.css]: Rename it to `colors.css`

File: client/components/AppHeader/AppHeader.css (hunk 1 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.css]: @rosendi fixed

File: client/components/AppHeader/AppHeader.css (hunk 8 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.css]: @rosendi fixed

File: client/components/AppHeader/AppHeader.css (hunk 7 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.css]: @rosendi fixed

File: client/components/AppHeader/AppHeader.css (hunk 2 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.css]: @rosendi fixed

File: client/components/AppHeader/AppHeader.js (hunk 4 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.js]: @rosendi fixed

File: client/components/AppHeader/AppHeaderItem.js (hunk 5 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeaderItem.js]: @rosendi fixed

File: client/styles/pallete.css (hunk 9 above)
Comment (zouxuoz) [inline client/styles/pallete.css]: @rosendi fixed

File: client/styles/pallete.css (hunk 6 above)
Comment (zouxuoz) [inline client/styles/pallete.css]: @rosendi fixed

File: client/components/AppHeader/AppHeader.css (hunk 3 above)
Comment (zouxuoz) [inline client/components/AppHeader/AppHeader.css]: I think we can use: `.app-header__item + .app-header__item`, it's more readable.

File: client/assets/logo.svg [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+  <text id="β" font-family="OpenSans, Open Sans" font-size="13" font-weight="normal">
```
Comment (rosendi) [inline client/assets/logo.svg]: Convert it to a path.

File: client/components/AppHeader/AppHeader.css [hunk 11]
```diff
@@ -0,0 +1,19 @@
+@import "styles/colors.css";
+
+.app-header {
+  background-color: var(--darkblueColor);
+  display: flex;
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.css]: I think we can remove the 5, 6 line and use `margin: 0 auto` in `.app-header__container`, can't we?

File: client/components/AppHeader/AppHeaderItem.css [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.app-header__link {
+  color: color(white a(60%));
+  text-decoration: none;
+  transition: color .2s ease;
+  white-space: nowrap;
+}
+
+.app-header__link:hover {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeaderItem.css]: For pseudo-classes you should nesting! `hover`,`focus`, etc...

```
.app-header__link {
    color: color(white a(60%));
    text-decoration: none;
    transition: color .2s ease;
    white-space: nowrap;

     &:hover {
       color: white;
    }
}
```

File: client/components/AppHeader/AppHeaderLogo.css [hunk 13]
```diff
@@ -0,0 +1,12 @@
+@import "styles/colors.css";
+
+.app-header__logo {
+  align-items: center;
+  display: flex;
+  fill: color(white a(60%));
+  margin-right: 24px;
+}
+
+.app-header__logo:hover {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeaderLogo.css]: Use nesting here!

File: client/layouts/AppLayout/AppLayout.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
+import React, { Component, PropTypes } from 'react';
+import AppHeader from 'components/AppHeader';
+
+import 'styles/index.css';
```
Comment (rosendi) [inline client/layouts/AppLayout/AppLayout.js]: remove it

File: client/components/AppHeader/AppHeaderItem.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+import './AppHeaderItem.css';
+
+export default AppHeaderItem = () => (
```
Comment (rosendi) [inline client/components/AppHeader/AppHeaderItem.js]: export default AppHeaderItem = ({ to, children }) => ...

**Pull request merged**